cli/cmd/check_extensions.go

```diff
@@ -0,0 +1,319 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/linkerd/linkerd2/pkg/version"
+	"github.com/mattn/go-isatty"
+	utilsexec "k8s.io/utils/exec"
+)
+
+// glob is satisfied by filepath.Glob.
+type glob func(string) ([]string, error)
+
+// extension contains the full path of an extension executable. If it's a
+// a built-in extension, path will be the `linkerd` executable and builtin will
+// be the extension name (jaeger, multicluster, or viz).
+type extension struct {
+	path    string
+	builtin string
+}
+
+var (
+	builtInChecks = map[string]struct{}{
+		"jaeger":       {},
+		"multicluster": {},
+		"viz":          {},
+	}
+)
+
+// findExtensions searches the path for all linkerd-* executables and returns a
+// slice of check commands, and a slice of missing checks.
+func findExtensions(pathEnv string, glob glob, exec utilsexec.Interface, nsLabels []string) ([]extension, []string) {
+	cliExtensions := findCLIExtensionsOnPath(pathEnv, glob, exec)
+
+	// first, collect config extensions that are "always" enabled
+	extensions, checksSeen := findAlwaysChecks(cliExtensions, exec)
+
+	// nsLabelSet keeps track of the extension namespaces that we are still
+	// searching for an executable for.
```

Comment:
Would it be accurate to say nsLabelSet is the set of extension names which are installed on the cluster but are not "always" checks?